XIVComboPlugin/IconReplacer.cs

```diff
@@ -904,6 +916,116 @@ private ulong GetIconDetour(byte self, uint actionID)
+                        return iconHook.Original(self, PCT.Bliz2);
+                    return iconHook.Original(self, PCT.Fire2);
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoHolyWhiteCombo))
+            {
+                if (actionID == PCT.HolyWhite)
+                {
+                    if (SearchBuffArray(PCT.Monochrome))
+                        return PCT.CometBlack;
+                    return PCT.HolyWhite;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoMotifMuseFeature))
+            {
+                if (actionID == PCT.CreatureMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.CreatureMotifDrawn)
+                        return iconHook.Original(self, PCT.LivingMuse);
+                    return iconHook.Original(self, actionID);
+                }
+
+                if (actionID == PCT.WeaponMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.WeaponMotifDrawn)
+                        return iconHook.Original(self, PCT.SteelMuse);
+                    if(SearchBuffArray(PCT.HammerReady))
+                        return iconHook.Original(self, PCT.HammerStamp);
+                    return iconHook.Original(self, actionID);
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoStarrySkyCombo))
+            {
+                if (actionID == PCT.LandscapeMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.LandscapeMotifDrawn)
+                        return PCT.StarryMuse;
+                    if (SearchBuffArray(PCT.StarStruck))
+                        return PCT.StarPrism;
+                    return PCT.StarryMotif;
+                }
+            }
+            
+            //VIPER
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperDeathRattleMaximum))
+            {
+                if (actionID == VPR.SteelFangs || actionID == VPR.DreadFangs)
+                {
+                    if (iconHook.Original(self, VPR.SerpentsTail) == VPR.DeathRattle)
+                        return VPR.DeathRattle;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperAoELash))
+            {
+
+                if (actionID == VPR.DreadMaw || actionID == VPR.SteelMaw)
+                {
+                    if (iconHook.Original(self, VPR.SerpentsTail) == VPR.LastLash)
+                        return VPR.LastLash;
+                }
+            }
+            
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperLegacyUnleashed)){
+                if (actionID == VPR.Reawaken && SearchBuffArray(VPR.Buffs.Reawakened))
```

Comment:
Steel Maw/Fangs already turn into the generational abilities, so it would make sense to put the Legacy abilities on top of those as well. This may be a case where it's better to do the actionID check before the config check. For now, take this combo out and I'll write it myself.

Reply:
Sounds good, I have commented out this block :)